certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -468,6 +468,44 @@ def test_reuse_key(context: IntegrationTestsContext) -> None:
     assert len({cert1, cert2, cert3}) == 3
 
 
+def test_reuse_key_allow_subset_of_names(context: IntegrationTestsContext) -> None:
+    """Test that key is reused when allow_subset_of_names is set."""
+    certname = context.get_domain('dns1')
+    domains = ','.join([certname, context.get_domain('fail-dns1')])
+    context.certbot([
+        '-a', 'manual', '-d', domains,
+        '--allow-subset-of-names',
+        '--preferred-challenges', 'dns',
+        '--manual-auth-hook', context.manual_dns_auth_hook_allow_fail,
+        '--manual-cleanup-hook', context.manual_dns_cleanup_hook,
+        '--reuse-key'
+    ])
+
+    stdout, _ = context.certbot(['certificates'])
+    assert context.get_domain('fail-dns1') in stdout
+    certname = context.get_domain('dns1')
+    with open(join(context.config_dir, 'archive/{0}/privkey1.pem').format(certname), 'r') as file:
+        privkey1 = file.read()
```

Comment:
i think this can be deleted since `privkey1` gets overwritten later w/ the same file